### ISSUE TITLE
[Feature suggestion] `.allowInsert()` and `.allowUpsert()` should be able to handle RelationExpression object notation

### ISSUE THREAD

**Issue by johannessjoberg** [Vincit/objection.js#1567]
Currently it's not possible to use `RelationExpression` object notation in `allowInsert()` and `.allowUpsert()` but it works well in `.allowEager()` and `.eager()`.

Example: `await Person.query().allowInsert({ pets: true }).insertGraph({ ... });` throws: `TypeError: allowedExpression.isSubExpression is not a function`

**Comment by koskimas** [comment]
Definitely! I thought they did handle that already :smile:

**Comment by johannessjoberg** [comment]
Awesome! 😀

**Comment by johannessjoberg** [comment]
I'm happy to supply a PR. Should it aimed to go into master, or 2.0 branch, or both?

**Comment by koskimas** [comment]
2.0 would be better. It will be the next version anyway. In 2.0 `allowInsert` and `allowUpsert` have been deprecated. There's only `allowGraph`. Actualy, this may already be fixed there.

**Comment by jamespedid** [comment]
I ran into this problem today. I switched to the string version, but I don't think it is working presently. Got an error that resembled this.

This is for version Objection 1.6.11 though.

```
TypeError: allowedExpression.isSubExpression is not a function
    at checkForUnallowedRelationErrors (/home/app/node_modules/objection/lib/queryBuilder/graph/GraphUpsert.js:214:28)
    at /home/app/node_modules/objection/lib/queryBuilder/graph/GraphUpsert.js:174:5
From previous event:
    at processImmediate (internal/timers.js:439:21)
From previous event:
    at QueryBuilder.execute (/home/app/node_modules/objection/lib/queryBuilder/QueryBuilder.js:580:8)
```

**Comment by johannessjoberg** [comment]
That's odd, string expression is working for us on `1.6.11`.

**Comment by jamespedid** [comment]
Sorry, I worded that poorly. The string version **is** working for me, but not the object version.

**Comment by koskimas** [comment]
This is actually already fixed in 2.0 branch.

**Comment by johannessjoberg** [comment]
Great! 👍
